### PR TITLE
[rework] inventory hud

### LINKED ISSUE #315: [Rework] Inventory HUD
- Icons but find from internet
- Animation for toggle
- 3 slots

### PR DESCRIPTION
**Pull request by rafiftaris**
### Main changes
- item slot hud
- item hud manager
- tween on item slot hud
- **bugfix** on item detector: item is not removed from nearby interactables after picked, creating duplicates when discarding

### Tasks List
- [close #315 ]

### Attachments
https://user-images.githubusercontent.com/51603162/167843419-be6424a2-9b45-485b-a433-9b536fb2d7a4.mp4

### New Packages
- [Unity-tweens](https://github.com/jeffreylanters/unity-tweens.git)

### CODE REVIEW

File: Assets/Astralization/Scripts/UI/Drawer/CircleGraphic.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+			vh.AddVert(new Vector3(cos * innerWidth + deltaWidth, sin * innerHeight + deltaHeight, 0f), color32, uv);
+
+			vh.AddTriangle(triangleIndex, triangleIndex - 2, triangleIndex - 1);
+			vh.AddTriangle(triangleIndex, triangleIndex - 1, triangleIndex + 1);
+		}
+
+		vh.AddTriangle(0, triangleIndex - 2, triangleIndex - 1);
+		vh.AddTriangle(0, triangleIndex - 1, 1);
+	}
+}
```
Comment (uciii) [inline Assets/Astralization/Scripts/UI/Drawer/CircleGraphic.cs:189]: EoF

**Review by naufalhilmi12345 — CHANGES_REQUESTED** [review]
LGTM for implementation. there is one question for the code

**Review by VadimBhirawa — CHANGES_REQUESTED** [review]
It already looks pretty good, but I would add some extras:
- Maybe add a function so that when the hud is expanded, players can press Tab to close the hud (some people would expand just to check what items they're bringing).
- The duration of the expansion (hud expand), is 2s, but should refresh when players change the active item slot (either via scroll or via quickslot).

File: Assets/Astralization/Scripts/Player/Inventory.cs [hunk 2]
```diff
@@ -41,9 +45,11 @@ public class Inventory : MonoBehaviour, IInventory
     [Tooltip("The list of items")]
     private Item[] _items;
 
+    [Range(3, 5)]
+    [SerializeField]
     [Tooltip("Inventory Length")]
-    [Range(3,5)]
-    public int InvenLength = 3;
+    private int _size = 3;
```
[the remark this answers is not in the document]
Comment (rafiftaris) [reply Assets/Astralization/Scripts/Player/Inventory.cs:51]: nama class udah inventory, kalo variable ada kata2 inven lagi jadi redundant menurut gua. Kalo mau akses variable itu, Inventory.Size lebih enak dibaca daripada Inventory.SizeInven kan?

File: Assets/Astralization/Scripts/Interactables/Item/Item.cs [hunk 3]
```diff
@@ -18,7 +22,7 @@ public abstract class Item : Interactable, IItem
     [Header("Item Logo")]
     [SerializeField]
     [Tooltip("The item logo for HUD")]
-    private Sprite _HudLogo;
+    protected Sprite _HudLogo;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Interactables/Item/Item.cs:25]: remove underscore

File: Assets/Astralization/Scripts/UI/ItemHud/ItemHudDisplay.cs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    [SerializeField]
+    private Vector3 _itemSlotGap = new Vector3(10f, 0f, 0f);
+    [SerializeField]
+    private Vector3 _itemSlotStartingPosition = new Vector3(1750f, 20f, 0f);
+    [SerializeField]
+    private float _tweenDuration = 0.5f;
+    [SerializeField]
+    private float _expandDuration = 3f;
+
+    private ItemSlot[] itemSlots;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/UI/ItemHud/ItemHudDisplay.cs:43]: add underscore on private

**Review by VadimBhirawa — APPROVED** [review]
It all looks good now!

**Review by naufalhilmi12345 — APPROVED** [review]
Cool! LGTM

**Pull request merged**